src/schema/v2/me/setOrUnsetUserFlagMutation.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  GraphQLNonNull,
+  GraphQLObjectType,
+  GraphQLString,
+  GraphQLUnionType,
+} from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import { GravityMutationErrorType } from "lib/gravityErrorHandler"
+import { snakeCase } from "lodash"
+import { ResolverContext } from "types/graphql"
+import { userFlagsResolver, UserFlagsType } from "./user_flags"
+
+const MutationSuccessType = new GraphQLObjectType<any, ResolverContext>({
+  name: "SetOrUnsetUserFlagMutationSuccess",
+  isTypeOf: (data) => {
+    return data._type !== "GravityMutationError"
+  },
+  fields: () => ({
+    userFlags: {
+      type: UserFlagsType,
+      resolve: async (root, options, loaders) => {
+        return await userFlagsResolver(root, options, loaders)
+      },
+    },
+  }),
+})
+
+const MutationFailureType = new GraphQLObjectType<any, ResolverContext>({
+  name: "SetOrUnsetUserFlagMutationFailure",
+  isTypeOf: (data) => {
+    return data._type === "GravityMutationError"
+  },
+  fields: () => ({
+    mutationError: {
+      type: GravityMutationErrorType,
+      resolve: (err) => (typeof err.message === "object" ? err.message : err),
+    },
+  }),
+})
+
+const SetOrUnsetUserFlagMutationType = new GraphQLUnionType({
+  name: "SetOrUnsetUserFlagMutationType",
+  types: [MutationSuccessType, MutationFailureType],
+})
+
+export const setOrUnsetUserFlagMutation = mutationWithClientMutationId<
+  any,
+  any,
+  ResolverContext
+>({
+  name: "SetOrUnsetUserFlagMutation",
```

Comment:
Suggestion: 

I would simplify the naming and just call the mutation setUserFlagMutation. If you pass null, the flag will be **set** as well.

Reply:
Thanks @MounirDhahri @olerichter00 . I agree the name seems some kinda way. The thing is how `user_flags` works is when the value is null, the key is completely deleted from the user_flags. It is not set per se.
So when we say `setUserFlagMutation` it may be misnomer because null and undefined cannot be set. Also `updateUserFlagMutation` too because the flag you are trying to update can be deleted.
I am open to a better name suggestion for this mutation 🙏